pkg/coscheduling/coscheduling.go

```diff
@@ -581,17 +514,45 @@ func (cs *Coscheduling) getBoundPods(podGroupName, namespace string, determined
 }
 
 func (cs *Coscheduling) getWaitingPods(namespace string) *v1.PodList {
-	pods, err := cs.frameworkHandle.ClientSet().CoreV1().Pods(namespace).List(context.TODO(), metav1.ListOptions{
+	fieldSelector := fmt.Sprintf("%s,%s", "status.phase=Pending",
+		fields.SelectorFromSet(fields.Set{"spec.nodeName": ""}).String())
+
+	podsList, err := cs.frameworkHandle.ClientSet().CoreV1().Pods(namespace).List(context.TODO(), metav1.ListOptions{
 		LabelSelector: "determined",
-		FieldSelector: "status.phase=Pending",
+		FieldSelector: fieldSelector,
 	})
 
 	if err != nil {
 		klog.Error(err)
 		return nil
 	}
 
-	return pods
+	sort.Slice(podsList.Items, func(i, j int) bool { // TODO: fix this
```

Comment:
What's the fix here, just factoring out the anonymous function?

Reply:
yeah so I wanted to use the same logic as they had in `Less()` but I couldn't just use it directly because the arg types were wrong. So I created a compare function and used that in both locations.